ntbs-service/DataMigration/ImportLogger.cs

```diff
@@ -1,69 +1,181 @@
-        void LogSuccess(PerformContext context, string requestId, string message);
-        void LogImportFailure(PerformContext context, string requestId, string message, Exception exception = null);
-        void LogError(PerformContext context, string requestId, string message, Exception exception = null);
+        void LogInformation(PerformContext context, int batchId, string message);
+
+        Task LogNotificationWarning(PerformContext context, int batchId, string legacyId, string message);
+        Task LogGroupWarning(PerformContext context, int batchId, IList<Notification> notificationGroup,
+            string message);
+
+        void LogSuccess(PerformContext context, int batchId, string message);
+        Task LogGroupSuccess(PerformContext context, int batchId, IList<Notification> notificationGroup);
+        Task LogImportGroupFailure(PerformContext context, int batchId, IList<Notification> notificationGroup,
+            string message, Exception exception = null);
+
+        Task LogNotificationError(PerformContext context, int batchId, string legacyId, string message);
+        Task LogGroupError(PerformContext context, int batchId, IList<Notification> notificationGroup,
+            string message);
     }
 
     public class ImportLogger : IImportLogger
     {
-        public void LogInformation(PerformContext context, string requestId, string message)
+        private readonly INotificationImportRepository _notificationImportRepository;
+
+        public ImportLogger(INotificationImportRepository notificationImportRepository)
         {
-            Log.Information($"NOTIFICATION IMPORT - {requestId} - {message}");
-            context.WriteLine($"NOTIFICATION IMPORT - {requestId} - {message}");
+            _notificationImportRepository = notificationImportRepository;
+        }
+
+        public void LogInformation(PerformContext context, int batchId, string message)
+        {
+            Log.Information($"NOTIFICATION IMPORT - {batchId} - {message}");
+            context.WriteLine($"NOTIFICATION IMPORT - {batchId} - {message}");
 
         }
-        public void LogSuccess(PerformContext context, string requestId, string message)
+        public void LogSuccess(PerformContext context, int batchId, string message)
         {
-            Log.Information($"NOTIFICATION IMPORT - {requestId} - {message}");
+            Log.Information($"NOTIFICATION IMPORT - {batchId} - {message}");
 
             context.SetTextColor(ConsoleTextColor.Green);
-            context.WriteLine($"NOTIFICATION IMPORT - {requestId} - {message}");
+            context.WriteLine($"NOTIFICATION IMPORT - {batchId} - {message}");
             context.ResetTextColor();
         }
 
-        public void LogImportFailure(PerformContext context, string requestId, string message, Exception exception = null)
+        public async Task LogGroupSuccess(PerformContext context, int batchId, IList<Notification> notificationGroup)

```

Comment:
There's some ambiguity in this review over the word "group". Here it means "Notification Group" but elsewhere it means "Batch Group". I wonder if we should rename the top level object to `LegacyImportMigrationRun` or something like that, to remove this ambiguity?